### PR TITLE
add simplified attribute in registry config

### PR DESCRIPTION
**Pull request by gaoxinge**
**What this PR does**:

- add simplified attribute in registry config

**Special notes for your reviewer**:

Adding a simplified attribute in registry config is for:

- Firstly, there is a simplified attribute in [java version](http://dubbo.apache.org/en-us/docs/user/references/xml/dubbo-registry.html).

- Secondly, simplified provider urls can release the pressure of the registry center. And we can move other parameters in provider ursl to metedata center and config center.

The code follows the [java version](https://github.com/apache/dubbo/blob/317e62f73fd0a9bac4b34eb8b2c31bbae591ef91/dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryProtocol.java#L204), but not strictly.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: common/url.go [hunk 1]
```diff
@@ -454,33 +455,28 @@ func (c URL) GetRawParam(key string) string {
...
 	}
 	return int64(r)
 }
 
 // GetMethodParamInt ...
 func (c URL) GetMethodParamInt(method string, key string, d int64) int64 {
-	var r int
-	var err error
 	c.paramsLock.RLock()
 	defer c.paramsLock.RUnlock()
```
Comment (zouyx) [inline common/url.go:477]: why do you add a read lock here?

``` func (c URL) GetParam(s string, d string) string ```  has lock too

File: common/url.go [hunk 2]
```diff
@@ -615,8 +609,29 @@ func (c *URL) Clone() *URL {
 	return newUrl
 }
 
+func (c *URL) CloneWithParams(reserveParams []string, methods []string) *URL {
```
Comment (zouyx) [inline common/url.go:612]: Should add a  comment for  exported method

Comment (gaoxinge) [reply common/url.go]: Add comment.

File: common/url.go (hunk 1 above)
Comment (gaoxinge) [reply common/url.go:475]: You are right. I restore the code.

Comment (gaoxinge) [reply common/url.go]: You are right. I remove the lock.

File: common/url.go [hunk 3]
```diff
@@ -615,8 +604,30 @@ func (c *URL) Clone() *URL {
 	return newUrl
 }
 
+// Copy url based on the reserved parameters' keys.
+func (c *URL) CloneWithParams(reserveParams []string) *URL {
+	params := url.Values{}
+	for _, reserveParam := range reserveParams {
+		v := c.GetParam(reserveParam, "")
+		if v != "" {
```
Comment (fangyincheng) [inline common/url.go:612]: len(v)==0

Comment (fangyincheng) [inline common/url.go:612]: len(v)==0

Comment (gaoxinge) [reply common/url.go]: Nice. I fixed this.

Comment (gaoxinge) [reply common/url.go]: This is same as above.

**Pull request merged**